src/kde/kde.hh

```diff
@@ -29,46 +28,97 @@ enum kde_coordinates { CART, CYL, SPH };
+
+  //! Calculate radius
+  inline double calc_radius(const std::array<double, 3> &center,
+                            const std::array<double, 3> &location) const;
+
+  //! Calculate Arch Length between two locations at a specific radius
+  inline double calc_arch_length(const std::array<double, 3> &center, const double radius,
+                                 const std::array<double, 3> &location_1,
+                                 const std::array<double, 3> &location_2) const;
+  //! Setup a spherical reconstruction with a
+  void set_sphere_center(const std::array<double, 3> &sph_center, const double min_radius,
+                         const double max_radius) {
+    Insist(max_radius > min_radius, "Spherical KDE max radius must be larger then min radius");
+
+    use_spherical_reconstruction = true;
+    sphere_center = sph_center;
+    sphere_min_radius = min_radius;
+    sphere_max_radius = max_radius;
+  }
 
 protected:
   // IMPLEMENTATION
 
 private:
-  // NESTED CLASSES AND TYPEDEFS
-
-  // IMPLEMENTATION
+  //! Private function to calculate kernel weight
+  double calc_weight(const std::array<double, 3> &r0, const std::array<double, 3> &one_over_h0,
+                     const std::array<double, 3> &r, const std::array<double, 3> &one_over_h,
+                     const quick_index &qindex, const double &discontinuity_cutoff) const {
+    return use_spherical_reconstruction
+               ? calc_spherical_weight(r0, one_over_h0, r, one_over_h, qindex, discontinuity_cutoff)
+               : calc_cartesian_weight(r0, one_over_h0, r, one_over_h, qindex,
+                                       discontinuity_cutoff);
+  }
+
+  void calc_win_min_max(const quick_index &qindex, const std::array<double, 3> &position,
+                        const std::array<double, 3> &one_over_bandwidth, std::array<double, 3> &min,
+                        std::array<double, 3> &max) const;
+
+  double calc_spherical_weight(const std::array<double, 3> &r0,
+                               const std::array<double, 3> &one_over_h0,
+                               const std::array<double, 3> &r,
+                               const std::array<double, 3> &one_over_h, const quick_index &qindex,
+                               const double &discontinuity_cutoff) const;
+
+  double calc_cartesian_weight(const std::array<double, 3> &r0,
+                               const std::array<double, 3> &one_over_h0,
+                               const std::array<double, 3> &r,
+                               const std::array<double, 3> &one_over_h, const quick_index &qindex,
+                               const double &discontinuity_cutoff) const;
 
   // DATA
+  //! reflecting boundary conditions [lower_x, upper_x, lower_y, upper_y, lower_z, upper_z]
+  const std::array<bool, 6> reflect_boundary;
+  //! Spherical Mesh Reconstruction Data
+  std::array<double, 3> sphere_center{0.0, 0.0, 0.0};
+  double sphere_min_radius{0.0};
```

Comment:
Are trailing comments needed for these variables?

Reply:
I think they are pretty descriptive but I could add some comments if you want.